### ISSUE TITLE
Support running tendermint as a separate process/container.

### ISSUE THREAD

**Issue by ruseinov** [blocklayerhq/chainkit#12]
Current implementation assumes that we're running everything in one container, which does not really resemble the reality and has it's limitations. 

It would be nice to be able to spin up tendermint as a separate container/app and the actual application would run in another one.

**Comment by samalba** [comment]
I don't really see the need until we can support some sort for shared security between abci apps and a single tendermint runtime.

Do you have an example project that would require tendermint to run separately?

**Comment by ruseinov** [comment]
yes, our weave apps rely on tendermint running elsewhere rather than the same process

**Comment by ruseinov** [comment]
The issue with the current approach is that it assumes that the app bundles terndermint.
Our apps communicate to tendermint via tcp socket and don't have tendermint runtime dependency.

**Comment by samalba** [comment]
I see, it makes sense. Thanks for the details.

After talking to many cosmos-sdk users, I think the most common approach is to bundle tendermint, but I definitely agree that running it separately should be supported as well, so I'll flag this as feature request.
